File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    tempArr = []
    
    # 문자열 s를 탐색하면서
    for i in range(len(s)):
        # 만약 s[i]번째 문자열이 tempArr에 없으면  
        if s[i] not in tempArr:
            tempArr.append(s[i]) # tempArr에 s[i]문자열을 추가하고
            answer.append(-1) # answer 배열에는 -1을 추가
        else: # 만약 이미 있다면
            lastIdx = s[:i].rindex(s[i]) # 탐색의 진행상황을 통해 원본 문자열으로 가서 마지막 인덱스 계산
            answer.append(i - lastIdx) # answer 배열에 i - lastIdx를 빼준다
    
    return answer

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    var answer = [];
    var tempArr = []; // 문자열을 담아줄 임시 열
    
    // 문자열 s를 순회하면서
    for (i = 0; i < s.length; i++) {
        if (!tempArr.includes(s[i])) {
            tempArr.push(s[i])
            answer.push(-1)
        } else {
            let lastIdx = s.lastIndexOf(s[i], i - 1);
            answer.push(i - lastIdx)
        }
    }
    
    return answer;
}